rs/rosetta-api/icrc1/ledger/src/lib.rs

```diff
@@ -326,6 +329,18 @@ pub enum LedgerArgument {
     Upgrade(Option<UpgradeArgs>),
 }
 
+const UPGRADES_MEMORY_ID: MemoryId = MemoryId::new(0);
+
+thread_local! {
+    static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> = RefCell::new(
+        MemoryManager::init(DefaultMemoryImpl::default())
```

Comment:
Doesn't this initialize the Memory manager in the post-upgrade of Canister version 1 corrupting the raw state written by StableWriter in Canister version 0?

Reply:
My guess would be that it is lazy initialized and we never use it if stable memory has raw state written to it.

Reply:
You're right. Thank you for the explanation!